src/astro/databases/__init__.py

```diff
@@ -0,0 +1,34 @@
+import importlib
+from pathlib import Path
+
+from airflow.hooks.base import BaseHook
+
+from astro.utils.path import get_dict_with_module_names_to_dot_notations
+
+DEFAULT_CONN_TYPE_TO_MODULE_PATH = get_dict_with_module_names_to_dot_notations(
+    Path(__file__)
+)
+CUSTOM_CONN_TYPE_TO_MODULE_PATH = {
+    "gcpbigquery": DEFAULT_CONN_TYPE_TO_MODULE_PATH["bigquery"],
+    "google_cloud_platform": DEFAULT_CONN_TYPE_TO_MODULE_PATH["bigquery"],
+}
+CONN_TYPE_TO_MODULE_PATH = {
+    **DEFAULT_CONN_TYPE_TO_MODULE_PATH,
+    **CUSTOM_CONN_TYPE_TO_MODULE_PATH,
+}
+SUPPORTED_DATABASES = set(DEFAULT_CONN_TYPE_TO_MODULE_PATH.keys())
+
+
+def get_database_from_conn_id(conn_id: str = ""):
+    """
+    Given a conn_id, return the associated Database class.
+    If null, the Database class defaults to the default Airflow connection, if it exists.
```

Comment:
I am not sure about this, how will this function know the default conn_type?

Reply:
Agreed, this feels like something that should be explicit

Reply:
@kaxil at the moment, it is up to the BaseDatabase subclass (e.g. `SqliteDatabase,` `BigqueryDatabase`)  to declare and use the default connection. I could change the interface for the connection parameter to be mandatory in the BaseDatabase instantiation method and have this class retrieve the attribute `DEFAULT_CONNECTION` from the module is defined.

@dimberman, I detailed the motivation in this other comment:
https://github.com/astro-projects/astro/pull/320#discussion_r859571663
This type of implicit behavior is already used by Airflow users when they instantiate hooks, for instance.

Reply:
@dimberman @kaxil I roll back my previous position, I'll remove the handling of default connection here. In this particular case, we need `conn_id`  to identify `conn_type` - as explained by @kaxil . I'll fix this.